library/src/scala/TupleXXL.scala

```diff
@@ -1,35 +1,35 @@
 package scala
 import java.util.Arrays.{deepEquals, deepHashCode}
 
-final class TupleXXL private (es: Array[Object]) extends Product {
+final class TupleXXL private (es: IArray[Object]) extends Product {
   assert(es.length > 22)
 
   def productElement(n: Int): Any = es(n)
   def productArity: Int = es.length
 
-  override def toString = elems.mkString("(", ",", ")")
-  override def hashCode = getClass.hashCode * 41 + deepHashCode(elems)
+  override def toString = elems.asInstanceOf[Array[Object]].mkString("(", ",", ")")
+  override def hashCode = getClass.hashCode * 41 + deepHashCode(elems.asInstanceOf[Array[Object]])
   override def canEqual(that: Any): Boolean = that match {
     case that: TupleXXL => that.productArity == this.productArity
     case _ => false
   }
 
   override def equals(that: Any) = that match {
-    case that: TupleXXL => deepEquals(this.elems, that.elems)
+    case that: TupleXXL => deepEquals(this.elems.asInstanceOf[Array[Object]], that.elems.asInstanceOf[Array[Object]])
     case _ => false
   }
-  def elems: Array[Object] = es
+  def elems: IArray[Object] = es
 
   def tailXXL: TupleXXL = {
     assert(es.length > 23)
-    new TupleXXL(es.tail)
+    new TupleXXL(es.asInstanceOf[Array[Object]].tail.asInstanceOf[IArray[Object]])
```

Comment:
It would be nice to support `IArray.tail`.

Reply:
Adding WrappedIArray would do it. Also imlementing  all ArrayOps on IArray would help